src/cadmium-dynamic-devstone.cpp

```diff
@@ -0,0 +1,141 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <iostream>
+#include <chrono>
+#include <algorithm>
+#include <fstream>
+
+#include <boost/program_options.hpp>
+
+#include <cadmium/engine/pdevs_dynamic_runner.hpp>
+
+#include "dynamic/LI_generator.cpp"
+#include "dynamic/HI_generator.cpp"
+#include "dynamic/HO_generator.cpp"
+#include "dynamic/HOmod_generator.cpp"
+
+namespace po=boost::program_options;
+using hclock=std::chrono::high_resolution_clock;
+using Time=float;
+
+int main(int argc, char* argv[]){
+    auto start = hclock::now();
+
+    // Declare the supported options.
+    po::options_description desc("Allowed options");
+    desc.add_options()
+            ("help", "produce help message")
+            ("kind", po::value<std::string>()->required(), "set kind of devstone: LI, HI, HO or HOmod")
+            ("width", po::value<int>()->required(), "set width of the DEVStone: integer value")
+            ("depth", po::value<int>()->required(), "set depth of the DEVStone: integer value")
+            ("int-cycles", po::value<int>()->required(), "set the Dhrystone cycles to expend in internal transtions: integer value")
+            ("ext-cycles", po::value<int>()->required(), "set the Dhrystone cycles to expend in external transtions: integer value")
+            ("time-advance", po::value<int>()->default_value(1), "set the time expend in external transtions by the Dhrystone in miliseconds: integer value")
+            ;
+
+    po::variables_map vm;
+    try {
+        po::store(po::parse_command_line(argc, argv, desc), vm);
+        po::notify(vm);
+    } catch ( boost::program_options::required_option be ){
+        if (vm.count("help")) {
+            std::cout << desc << "\n";
+            return 0;
+        } else {
+            std::cout << be.what() << std::endl;
+            std::cout << std::endl;
+            std::cout << "for mode information run: " << argv[0] << " --help" << std::endl;
+            return 1;
+        }
+    }
+    std::string kind = vm["kind"].as<std::string>();
+    if (kind.compare("LI") != 0  && kind.compare("HI") != 0 &&
+        kind.compare("HO") != 0 && kind.compare("HOmod") != 0) {
+        std::cout << "The kind needs to be LI, HI, HO or HOmod and received value was: " << kind << std::endl;
+        std::cout << "for mode information run: " << argv[0] << " --help" << std::endl;
+        return 1;
+    }
+
+    int width = vm["width"].as<int>();
+    int depth = vm["depth"].as<int>();
+    int int_cycles = vm["int-cycles"].as<int>();
+    int ext_cycles = vm["ext-cycles"].as<int>();
+    int time_advance = vm["time-advance"].as<int>();
+    //finished processing input
```

Comment:
I would add the kind as a "enum class" here so you don't have to deal with strings afterward.

Reply:
Maybe above where you ask it is none of these, you can use that as a "default case" after evaluated the 4 valid ones and assigning to the enum class.